hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerConfiguration.java

```diff
@@ -1693,6 +1726,14 @@ public boolean getIntraQueuePreemptionDisabled(String queue,
             + QUEUE_PREEMPTION_DISABLED, defaultVal);
   }
 
+  public void setDisablePreemptionForPreemtionModes(String preemptionType, boolean value) {
+    setBoolean(PREEMPTION_CONFIG_PREFIX + preemptionType, value);
+  }
```

Comment:
The property observe only is a valid one, however it's not connected to disabling preemption itself, so the naming is misleading. And observe_only is a property of preemption, I wouldn't call it a type. Also please don't expect the property name as a parameter, because it destroys the purpose of the setter.